spec/services/trainees/filter_spec.rb

```diff
@@ -22,6 +22,18 @@ module Trainees
       it { is_expected.not_to include(empty_trainee) }
     end
 
+    context "when HESA TRN data trainee exists" do
+      let!(:hesa_trn_data_trainee) { create(:trainee, record_source: RecordSources::HESA_TRN_DATA) }
+
+      it { is_expected.not_to include(hesa_trn_data_trainee) }
+    end
+
+    context "when a HESA collection trainee exists" do
+      let!(:hesa_collection_trainee) { create(:trainee, record_source: RecordSources::HESA_COLLECTION) }
+
+      it { is_expected.to include(hesa_collection_trainee) }
+    end
+
```

Comment:
Should we add a spec for `record_source: nil`?

Reply:
I was going to and then i thought.. meh I'll have to remove it when the validation goes in. But now I've written that sentence I disagree with myself. I'll add one.

I've also made it so that all the seeds have nil record_source to reflect prod this time 👍

Reply:
I guess all the other trainees in this test file have nil record source.. I'll still add one to be explicit.

Reply:
https://github.com/DFE-Digital/register-trainee-teachers/pull/2653/commits/f2f75093e9880482b8237e1388f0393645df90ab